### DISCUSSION TITLE
cannot execute binary file: Exec format error

### DISCUSSION THREAD

**Discussion by swhoro** [mbround18/valheim-docker#524] [Help!] [answered]
I install docker on openwrt x86_64, and it seems steamcmd doesn't work properly

my log:
[Valheim][root] : ###########################################################################
[Valheim][root] : Valheim Server - Wed 16 Feb 2022 04:30:35 PM CST
[Valheim][root] : Initializing your container...
[Valheim][root] : You must be in development. Good luck!
[Valheim][root] : ###########################################################################
[Valheim][root] : Switching UID and GID
[Valheim][root] : usermod: no changes
[Valheim][root] : 
[Valheim][root] : Setting up file systems
[Valheim][root] : Navigating to steam home...
[Valheim][root] : Launching as steam...
[Valheim][steam]: ###########################################################################
[Valheim][steam]: Valheim Server - Wed 16 Feb 2022 04:30:37 PM CST
[Valheim][steam]: STEAM_UID 1000 - STEAM_GUID 1000
[Valheim][steam]: Installing Valheim via odin 1.4.3...
[Valheim][steam]: ###########################################################################
[Valheim][steam]: Variables loaded.....
[Valheim][steam]: Port: 2456
[Valheim][steam]: Name: Created With Valheim Docker
[Valheim][steam]: World: Dedicated
[Valheim][steam]: Public: 1
[Valheim][steam]: Password: (REDACTED)
[Valheim][steam]: Running Install...
[ODIN][INFO]  - Installing 896660 to /home/steam/valheim
[ODIN][INFO]  - steamcmd found in path
/home/steam/.steam/steamcmd/steamcmd.sh: line 37: /home/steam/.steam/steamcmd/linux32/steamcmd: cannot execute binary file: Exec format error
[ODIN][INFO]  - Exited with http code: 126
cp: cannot stat '/home/steam/steamcmd/linux64/steamclient.so': No such file or directory
[Valheim][steam]: Initializing Variables....
[ODIN][ERROR] - Failed to find server executable! Please run `odin install`

**Comment by mbround18** [comment, marked as answer]
@swhoro wut O.o try adding this to your docker-compose for the valheim server

```yaml

platform: linux/amd64
```

**Reply by swhoro** [reply]
> @swhoro wut O.o try adding this to your docker-compose for the valheim server
> 
> ```yaml
> platform: linux/amd64
> ```

have the same error
here is my compose file:
![image](https://user-images.githubusercontent.com/34229589/154420963-80e9bf0d-b13f-4ba0-a2c6-294814566590.png)

**Reply by mbround18** [reply]
Could you try and change the image to be `mbround18/valheim:1.5.0` ?

**Reply by mbround18** [reply]
Additionally, seeing you are on OpenWRT for your router? how much memory / cpu does this system have? this image requires a min of 4gigs of ram to run.

**Reply by swhoro** [reply]
> Additionally, seeing you are on OpenWRT for your router? how much memory / cpu does this system have? this image requires a min of 4gigs of ram to run.

In fact, I install openwrt on proxmox ve, and the openwrt has 1 sockets 2 cores and 2GB
Yesterday before I opened the discussion, the openwrt has 5000MB memory. But I changed to 2048MB later.
I will try changing it to 5000MB later.
What is more , it seems 1.5.0 is the latest version of this docker image?

**Reply by mbround18** [reply]
1.5.0 is the latest version just for debugging purposes its easier for me to have peeps on a listed version :P ran into some issues where people were on the latest tag but were confused why it wasnt always up to date. Tbh, having you use 1.5.0 skips the conversation of comparing image sha's. 

The error you listed is concerning if you change to 1.5.0 and still see it, it would lead me to believe that the openwrt operating system might have something in it that isnt allowing the image to run the 32bit instruction set steam requires.

**Reply by swhoro** [reply]
> 1.5.0 is the latest version just for debugging purposes its easier for me to have peeps on a listed version :P ran into some issues where people were on the latest tag but were confused why it wasnt always up to date. Tbh, having you use 1.5.0 skips the conversation of comparing image sha's.
> 
> The error you listed is concerning if you change to 1.5.0 and still see it, it would lead me to believe that the openwrt operating system might have something in it that isnt allowing the image to run the 32bit instruction set steam requires.

I have changed to 1.5.0 and it has the same error.
I will try installing a debian or ubuntu on pve and using this docker image later.

**Reply by mbround18** [reply]
Please keep me posted on the outcome. I use ubuntu latest for my server with docker engine >=20 with no issue and would like to help you resolve this to get yall online.

**Reply by swhoro** [reply]
> Please keep me posted on the outcome. I use ubuntu latest for my server with docker engine >=20 with no issue and would like to help you resolve this to get yall online.

Thanks for your help.
Now I have run valheim server on ubuntu-server!

**Reply by swhoro** [reply]
I have another question, could you please help me?
In the past, I can use debugmode in valheim to cheat, such as press z to fly.
After I use a dedicated server, is there anyway to cheat?

**Reply by mbround18** [reply]
@swhoro my go to is 

https://www.nexusmods.com/valheim/mods/8

**Reply by swhoro** [reply]
> @swhoro my go to is
> 
> https://www.nexusmods.com/valheim/mods/8

Thank you for your reply!
